### PR TITLE
Cragify core test subfolders

### PR DESCRIPTION
**Pull request by dpeng817**

### CODE REVIEW

**Comment by dpeng817** [comment]
Current dependencies on/for this PR:
* master
  * **PR #9749** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9749" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/9749?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/reconstruct.py [hunk 1]
```diff
@@ -260,7 +260,9 @@ def subset_for_execution(
     ) -> "ReconstructablePipeline":
         # take a list of unresolved selection queries
         check.opt_list_param(solid_selection, "solid_selection", of_type=str)
-        check.opt_set_param(asset_selection, "asset_selection", of_type=AssetKey)
+        asset_selection = frozenset(
+            check.opt_set_param(asset_selection, "asset_selection", of_type=AssetKey)
+        )
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/reconstruct.py:265]: this is a behavior change - before `None` would flow through now we coerce in to empty set. Pretty sure None vs empty set is meaningful so dont think we want this change.

File: python_modules/dagster/dagster/_core/execution/execution_result.py [hunk 2]
```diff
@@ -39,7 +39,9 @@ def run_id(self) -> str:
     @property
     def success(self) -> bool:
         """bool: Whether execution was successful."""
-        return self.dagster_run.is_success
+        return self.dagster_run.is_success and not self.filter_events(
+            lambda evt: evt.is_step_failure or evt.is_run_failure
+        )
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/execution/execution_result.py:44]: this seems extremely dubious - how is the run successful and the events failures?

**Review by alangenfeld — CHANGES_REQUESTED** [review]
Definitely want behavior changes separated out from mass conversion PRs. 

PR summary should include how the mass modification was performed.

**Pull request closed without merging**